pandera/backends/pandas/components.py

```diff
@@ -95,14 +100,24 @@ def validate_column(check_obj, column_name):
                         check_obj[column_name].iloc[:, [i]], column_name
                     )
             else:
-                validate_column(check_obj, column_name)
+                if hasattr(schema, "drop_invalid") and schema.drop_invalid:
+                    # replace the check_obj with the validated check_obj
+                    check_obj = validate_column(
+                        check_obj, column_name, return_check_obj=True
+                    )
+                else:
+                    validate_column(check_obj, column_name)
 
         if lazy and error_handler.collected_errors:
-            raise SchemaErrors(
-                schema=schema,
-                schema_errors=error_handler.collected_errors,
-                data=check_obj,
-            )
+            if hasattr(schema, "drop_invalid") and schema.drop_invalid:
+                check_obj = self.drop_invalid_data(check_obj, error_handler)
+                return check_obj
```

Comment:
weird, it seems like the `test_drop_invalid_for_dataframe_schema` test should cover this case no? Is this a codecov bug?

Reply:
The hasattr? That was just some defensive code as there was a recent issue with the new default attr not being available on pickled schemas. Can drop if you think it is overkill

Reply:
no, it's okay to have that in there, it's just that codecov is complaining that this part of the code wasn't executed, meaning that lines 113 weren't executed during CI. But it seems like `test_drop_invalid_for_dataframe_schema` should have caused this code to run no?

Reply:
Ah no it doesn't appear so locally! Codecov saving us 😆 

This was my bad. So this code in this file:
```python
            if is_table(check_obj[column_name]):
                for i in range(check_obj[column_name].shape[1]):
                    validate_column(
                        check_obj[column_name].iloc[:, [i]], column_name
                    )
            else:
                if hasattr(schema, "drop_invalid") and schema.drop_invalid:
                    # replace the check_obj with the validated check_obj
                    check_obj = validate_column(
                        check_obj, column_name, return_check_obj=True
                    )
                else:
                    validate_column(check_obj, column_name)

        if lazy and error_handler.collected_errors:
            if hasattr(schema, "drop_invalid") and schema.drop_invalid: # the line in question!
                check_obj = self.drop_invalid_data(check_obj, error_handler)
                return check_obj
            else:
                raise SchemaErrors(
                    schema=schema,
                    schema_errors=error_handler.collected_errors,
                    data=check_obj,
                )
 ```
Raises an error on `validate_column` but does not rescue it. The rescue is done higher up in `ArraySchemaBackend.validate()`. This is now obvious as there is no `except` between `validate_column` and ` if lazy and error_handler.collected_errors`!

I've removed these lines and pushed the changes👌